File: app/src/main/java/com/decode/tumblr/posts/MainActivity.java
package com.decode.tumblr.posts;

import android.os.Bundle;
import android.view.Window;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.ViewModelProviders;

import com.decode.tumblr.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.toolbar)
    Toolbar toolbar;
    @BindView(R.id.txt_blog_title)
    TextView txtBlogTitle;
    @BindView(R.id.txt_total_posts)
    TextView txtTotalPosts;
    @BindView(R.id.txt_updated)
    TextView txtUpdated;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getWindow().requestFeature(Window.FEATURE_ACTION_BAR_OVERLAY);
        setContentView(R.layout.activity_main);
        setSupportActionBar(toolbar);
        ButterKnife.bind(this);

        // Set title
        toolbar.setTitle(getString(R.string.android_onboarding_project));

        PostViewModel postViewModel = ViewModelProviders.of(this).get(PostViewModel.class);
        postViewModel.fetchPosts();

        // Set header data
        postViewModel.getHeader().observe(this, header -> {
            if (header != null) {
                txtBlogTitle.setText(header.getTitle());
                txtTotalPosts.setText(header.getTotalPost());
                txtUpdated.setText(header.getUpdated());
            }
        });
    }
}

File: app/src/main/java/com/decode/tumblr/posts/MainActivity.kt
package com.decode.tumblr.posts

import android.os.Bundle
import android.view.Window
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.decode.tumblr.R
import kotlinx.android.synthetic.main.activity_main.*
import org.koin.android.viewmodel.ext.android.viewModel

class MainActivity : AppCompatActivity() {

    // Lazy Inject ViewModel
    val postViewModel: PostViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        window.requestFeature(Window.FEATURE_ACTION_BAR_OVERLAY)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        postViewModel.fetchPosts()

        // Set title
        toolbar!!.title = getString(R.string.android_onboarding_project)

        // Set header data
        postViewModel.header.observe(this, Observer { header ->
            if (header != null) {
                txt_blog_title!!.text = header.title
                txt_total_posts!!.text = header.totalPost
                txt_updated!!.text = header.updated
            }
        })
    }
}
